Bugfix/images

danielcherubini opened the pull request.

danielcherubini commented:
@inventor7777 here you go

https://github.com/danielcherubini/elegoo-homeassistant/releases/tag/v1.4.6b1

<img width="484" height="786" alt="image" src="https://github.com/user-attachments/assets/97d35952-eb77-4cf5-9905-227054468c86" />

<img width="909" height="1246" alt="image" src="https://github.com/user-attachments/assets/52064395-b5ec-45c8-bb21-952121163f86" />

danielcherubini commented:
When this runs.. please enable debug logging and paste me the logs

In the Elegoo Printer page
<img width="824" height="642" alt="image" src="https://github.com/user-attachments/assets/e12d8f8d-43b1-41b3-8466-834c6e84766d" />

danielcherubini commented:
I think 1.4.6b2 will do it...

The issue seems to be the "Unavailable" .. which is an availability check that the printer does before trying to get the image from the printer.. problem is that function is a synchronous function, and a lot of this lib has moved to async.

So now im just checking IF the printer is printing lol

danielcherubini commented:
You're going to have to be actively printing to test this.

inventor7777 commented:
Okay so interesting. I simply enabled proxy and the video stream is working, but still no cover image.

and I haven’t even changed my slicer WiFi settings (but should be fine as there would only be 4 connections total right?)


Log -
2025-07-10 18:12:29.887 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxying video request to http://192.168.50.123:3031/video
2025-07-10 18:12:30.319 ERROR (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Error proxying video stream: Cannot write to closing transport
2025-07-10 18:12:31.580 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxying video request to http://192.168.50.123:3031/video
2025-07-10 18:12:31.864 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.040 seconds (success: True)
2025-07-10 18:12:34.060 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.260 seconds (success: True)
2025-07-10 18:12:34.912 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxying video request to http://192.168.50.123:3031/video
2025-07-10 18:12:35.333 ERROR (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Error proxying video stream: Cannot write to closing transport
2025-07-10 18:12:36.079 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.059 seconds (success: True)
2025-07-10 18:12:38.703 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.912 seconds (success: True)
2025-07-10 18:12:40.821 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.219 seconds (success: True)
2025-07-10 18:12:44.797 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.006 seconds (success: True)
2025-07-10 18:12:48.792 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:12:50.618 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxying video request to http://192.168.50.123:3031/video
2025-07-10 18:12:50.977 ERROR (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Error proxying video stream: Cannot write to closing transport
2025-07-10 18:12:52.808 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:12:54.592 ERROR (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Error proxying video stream: Cannot write to closing transport
2025-07-10 18:12:54.912 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxying video request to http://192.168.50.123:3031/video
2025-07-10 18:12:55.295 ERROR (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Error proxying video stream: Cannot write to closing transport
2025-07-10 18:12:56.823 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.018 seconds (success: True)
2025-07-10 18:12:57.740 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxying video request to http://192.168.50.123:3031/video
2025-07-10 18:12:58.968 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxying video request to http://192.168.50.123:3031/video
2025-07-10 18:12:59.302 ERROR (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Error proxying video stream: Cannot write to closing transport
2025-07-10 18:13:00.793 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:13:04.794 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:13:08.794 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:13:12.795 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:13:14.056 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxying video request to http://192.168.50.123:3031/video
2025-07-10 18:13:14.457 ERROR (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Error proxying video stream: Cannot write to closing transport
2025-07-10 18:13:16.793 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:13:20.795 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:13:24.795 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:13:27.805 ERROR (MainThread) [homeassistant.components.mqtt.number] Invalid value for number.air_purifier_custom: 0 (range 1.0 - 13.0)
2025-07-10 18:13:29.267 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.016 seconds (success: True)
2025-07-10 18:13:32.794 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:13:36.798 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.007 seconds (success: True)
2025-07-10 18:13:40.793 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:13:44.793 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:13:48.795 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:13:52.803 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.004 seconds (success: True)
2025-07-10 18:13:56.793 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:13:58.535 ERROR (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] Error proxying video stream: 
2025-07-10 18:14:00.794 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:14:04.794 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:14:08.793 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:14:12.805 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:14:16.794 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)

inventor7777 commented:
And I’m printing again lol not even because of this but because I needed to 😂

danielcherubini commented:
You won't need to restart the printer for this. 

It's just a "maybe that's a good idea" thing. 

But.. great to hear about the video feed... Bad to hear about the image.. you're on 1.4.6b2?

inventor7777 commented:
Oddly enough, HA itself says 1.4.5 but HACS says 1.4.6b2 🤣

inventor7777 commented:
Why does the video feed only work in Proxy mode? Is that intended behavior?

inventor7777 commented:
Are we moving the discussion about the light control here as well?

inventor7777 commented:
I will try again.

inventor7777 commented:
what extra logs should I be looking for?

danielcherubini commented:
Why not. .. I haven't changed anything there.

danielcherubini commented:
> what extra logs should I be looking for?

Same debug logs.

inventor7777 commented:
I tried redownloading again. HA still shows the version as 1.4.5 but HACS shows 1.4.6b2


Logs
2025-07-10 18:31:25.853 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:31:27.849 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:31:29.404 INFO (MainThread) [custom_components.elegoo_printer] Closing connection to printer
2025-07-10 18:31:29.405 DEBUG (MainThread) [custom_components.elegoo_printer] WebSocket listener cancelled.
2025-07-10 18:31:29.405 INFO (MainThread) [custom_components.elegoo_printer] WebSocket listener stopped.
2025-07-10 18:31:29.415 INFO (SyncWorker_5) [custom_components.elegoo_printer] Stopping proxy server...
2025-07-10 18:31:29.429 INFO (Thread-2 (_start_servers_in_thread)) [custom_components.elegoo_printer] WebSocket client disconnected from 192.168.50.59
2025-07-10 18:31:29.449 INFO (SyncWorker_5) [custom_components.elegoo_printer] Proxy server stopped.
2025-07-10 18:31:29.452 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 1.606 seconds (success: True)
2025-07-10 18:31:29.454 DEBUG (MainThread) [custom_components.elegoo_printer] CONFIGURATION {'brand': 'ELEGOO', 'connection': '979d4C788A4a78bC777A870F1A02867A', 'firmware': 'V1.1.29', 'id': '0c6817140107202800007c0000000000', 'ip_address': '192.168.50.123', 'model': 'Centauri Carbon', 'name': 'Centauri Carbon', 'printer_type': 'fdm', 'protocol': 'V3.0.0', 'proxy_enabled': True}
2025-07-10 18:31:29.455 DEBUG (MainThread) [custom_components.elegoo_printer] Proxy server is enabled, attempting to create proxy server.
2025-07-10 18:31:29.469 INFO (MainThread) [custom_components.elegoo_printer] Initializing proxy server for remote printer 192.168.50.123
2025-07-10 18:31:29.486 INFO (Thread-99 (_start_servers_in_thread)) [custom_components.elegoo_printer] Main HTTP/WebSocket Proxy running on http://192.168.50.59:3030
2025-07-10 18:31:29.497 INFO (Thread-99 (_start_servers_in_thread)) [custom_components.elegoo_printer] Video Proxy running on http://192.168.50.59:3031
2025-07-10 18:31:29.504 INFO (Thread-99 (_start_servers_in_thread)) [custom_components.elegoo_printer] Discovery Proxy listening on UDP port 3000
2025-07-10 18:31:29.510 INFO (Thread-99 (_start_servers_in_thread)) [custom_components.elegoo_printer] Unified HTTP/WebSocket Proxy running on http://192.168.50.59:3030
2025-07-10 18:31:29.512 INFO (MainThread) [custom_components.elegoo_printer] Proxy server has started successfully.
2025-07-10 18:31:29.517 DEBUG (MainThread) [custom_components.elegoo_printer] Connecting to printer: Centauri Carbon at 192.168.50.59 with proxy enabled True and printer.proxy: True
2025-07-10 18:31:29.518 INFO (MainThread) [custom_components.elegoo_printer] Closing connection to printer
2025-07-10 18:31:29.518 DEBUG (MainThread) [custom_components.elegoo_printer] Connecting to printer: Centauri Carbon at 192.168.50.59 proxy_enabled: True
2025-07-10 18:31:29.518 INFO (MainThread) [custom_components.elegoo_printer] Client connecting to WebSocket at: ws://192.168.50.59:3030/websocket
2025-07-10 18:31:29.545 INFO (Thread-99 (_start_servers_in_thread)) [custom_components.elegoo_printer] WebSocket client connected from 192.168.50.59
2025-07-10 18:31:29.577 INFO (MainThread) [custom_components.elegoo_printer] Client successfully connected to: Centauri Carbon, via proxy: True
2025-07-10 18:31:29.577 INFO (MainThread) [custom_components.elegoo_printer] Polling Started
2025-07-10 18:31:29.582 INFO (Thread-99 (_start_servers_in_thread)) [custom_components.elegoo_printer] Proxy connected to remote printer WebSocket at 192.168.50.123
2025-07-10 18:31:31.581 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.004 seconds (success: True)
2025-07-10 18:31:31.582 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 23 sensor entities
2025-07-10 18:31:31.623 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 1 image entities
2025-07-10 18:31:31.634 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 1 light entities
2025-07-10 18:31:31.637 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 3 button entities
2025-07-10 18:31:33.589 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.005 seconds (success: True)
2025-07-10 18:31:37.780 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:31:41.265 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [281473087869376] Error handling message: Unknown error (unknown_error) Gabriel Baird from fe80::6e:95e5:de1:6b74 (Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/18.4 Safari/605.1.15)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 28, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 599, in handle_manifest_list
    raise int_or_exc
homeassistant.loader.IntegrationNotFound: Integration 'ble_monitor' not found.
2025-07-10 18:31:41.778 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:31:45.781 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-10 18:31:49.783 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:31:53.779 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-10 18:31:57.779 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)

danielcherubini commented:
https://github.com/danielcherubini/elegoo-homeassistant/releases/tag/v1.4.6b3

try this @inventor7777

danielcherubini commented:
ok, date no image... that tells me its now available. but the get the image isn't working.. theres missing debug logs there.. which tells me for some reason, your home assistant isn't trying to get the image. 

Which is weird

danielcherubini commented:
i'll have to look at this later

inventor7777 commented:
Something weird is going on now. 

The integration was not reporting anything this morning until I disabled the proxy, then it started working again.
![IMG_1652](https://github.com/user-attachments/assets/13ae5dea-eada-43bd-a8d4-8e288a53d1b7)
![IMG_1654](https://github.com/user-attachments/assets/f1f0e94b-0b9d-4590-9179-f666d7c2eb94)

inventor7777 commented:
![ABC9F0C3-1BAA-4BEA-AFB5-92419DD23A98_1_201_a](https://github.com/user-attachments/assets/4e6506c7-21a5-411b-8cdf-8ae9e4d9f9b4)

danielcherubini commented:
Strange.. so far you might be the only one with that.. remember the printer has a max connections, so once you use it directly in other places you can trip that and weird things happen.

That said.. i need to concentrate on this image problem first.

inventor7777 commented:
Gotcha! Well I’m usually pretty active around and if/when you need anything else once you get back home to help get this integration to its full potential I’ll be glad to do whatever ;)

danielcherubini commented:
I'll be mostly writing tests so I canwork on this remotely more reliably. Make Dev work much easier if I can spin up a fake printer.

inventor7777 commented:
Yes, definitely. Better than risking your own printer/materials for a HA integration!

danielcherubini commented:
@inventor7777 im about to release a new pre-release

danielcherubini commented:
https://github.com/danielcherubini/elegoo-homeassistant/releases/tag/v1.4.6b5

bystand3r commented:
Lights been like a rocket here today ! Super responsive.
image is showing date but not the image.


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: inventor7777 ***@***.***>
Sent: Saturday, July 12, 2025 8:59:58 PM
To: danielcherubini/elegoo-homeassistant ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [danielcherubini/elegoo-homeassistant] Bugfix/images (PR #156)

[https://avatars.githubusercontent.com/u/142270525?s=20&v=4]inventor7777 left a comment (danielcherubini/elegoo-homeassistant#156)<https://github.com/danielcherubini/elegoo-homeassistant/pull/156#issuecomment-3065315625>

@danielcherubini<https://github.com/danielcherubini> i haven’t printed anything yet as I have to go do something this morning (08:00 here rn) but the chamber light is now working and it’s very fast!!

Once I get back home and print something I’ll let you know how the cover image works.

—
Reply to this email directly, view it on GitHub<https://github.com/danielcherubini/elegoo-homeassistant/pull/156#issuecomment-3065315625>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AE5PW3UJR3B73EKF2PIEE7T3IEBE5AVCNFSM6AAAAACBIHNRU2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTANRVGMYTKNRSGU>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

danielcherubini commented:
Ok great. This is a step in the right direction that means now the problem is getting the image from the printer. 

The issue there is that the printer doesn't send the file type in the headers. So when home assistant does the get, it errors. I've written around this, but I'll actually factor that bug into my fake printer server I've made. So it mimics that crappy behaviour

danielcherubini commented:
https://github.com/danielcherubini/elegoo-homeassistant/releases/tag/v1.4.6b6

Try this. I think this might do the trick

inventor7777 commented:
I’m about to try it.

some more interesting behavior I have noticed is that the Centauri light CANNOT be controlled wirelessly once the print is running. Even Elegoo’s own Web UI says “modification failed”. So that might have been part of my issue as I was trying to change the light while printing. 

This could be related to the Timelapse but the printer screen controls it fine.

inventor7777 commented:
The cover image still won’t load yet I don’t see any special logs hmm 

do I need proxies enabled for cover image?

2025-07-12 17:54:56.910 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 1.306 seconds (success: True)
2025-07-12 17:54:56.952 DEBUG (MainThread) [custom_components.elegoo_printer] CONFIGURATION {'brand': 'ELEGOO', 'connection': '979d4C788A4a78bC777A870F1A02867A', 'firmware': 'V1.1.29', 'id': '0c6817140107202800007c0000000000', 'ip_address': '192.168.50.123', 'model': 'Centauri Carbon', 'name': 'Centauri Carbon', 'printer_type': 'fdm', 'protocol': 'V3.0.0', 'proxy_enabled': False}
2025-07-12 17:54:56.953 DEBUG (MainThread) [custom_components.elegoo_printer] Connecting to printer: Centauri Carbon at 192.168.50.123 with proxy enabled False and printer.proxy: False
2025-07-12 17:54:56.953 INFO (MainThread) [custom_components.elegoo_printer] Closing connection to printer
2025-07-12 17:54:56.953 DEBUG (MainThread) [custom_components.elegoo_printer] Connecting to printer: Centauri Carbon at 192.168.50.123 proxy_enabled: False
2025-07-12 17:54:56.953 INFO (MainThread) [custom_components.elegoo_printer] Client connecting to WebSocket at: ws://192.168.50.123:3030/websocket
2025-07-12 17:54:59.278 INFO (MainThread) [custom_components.elegoo_printer] Client successfully connected to: Centauri Carbon, via proxy: False
2025-07-12 17:54:59.278 INFO (MainThread) [custom_components.elegoo_printer] Polling Started
2025-07-12 17:55:02.089 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.811 seconds (success: True)
2025-07-12 17:55:02.090 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 23 sensor entities
2025-07-12 17:55:02.151 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 1 image entities
2025-07-12 17:55:02.173 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 1 light entities
2025-07-12 17:55:02.176 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 3 button entities
2025-07-12 17:55:04.053 ERROR (MainThread) [homeassistant.components.mjpeg] Error getting new camera image from Centauri Carbon Camera: All connection attempts failed
2025-07-12 17:55:04.162 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.070 seconds (success: True)
2025-07-12 17:55:06.046 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.014 seconds (success: True)
2025-07-12 17:55:07.950 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.024 seconds (success: True)
2025-07-12 17:55:09.927 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.007 seconds (success: True)
2025-07-12 17:55:11.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:13.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:55:15.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:55:16.931 WARNING (MainThread) [custom_components.mbapi2020.websocket] WSS Connection blocked: 429, message='Invalid response status', url='wss://websocket.amap-prod.mobilesdk.mercedes-benz.com/v2/ws', retry in 90 seconds...
2025-07-12 17:55:17.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:19.931 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.014 seconds (success: True)
2025-07-12 17:55:21.930 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:55:23.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-12 17:55:25.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-12 17:55:28.394 WARNING (MainThread) [homeassistant.helpers.service] Referenced entities light.rgbicw_workbench are missing or not currently available
2025-07-12 17:55:28.478 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.559 seconds (success: True)
2025-07-12 17:55:29.072 ERROR (MainThread) [homeassistant.components.mqtt.number] Invalid value for number.air_purifier_custom: 0 (range 1.0 - 13.0)
2025-07-12 17:55:30.456 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.012 seconds (success: True)
2025-07-12 17:55:31.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:33.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:55:35.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:37.918 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:39.918 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:41.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:55:43.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:55:45.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:47.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:49.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:51.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:55:53.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:55:55.918 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:57.918 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:55:59.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:56:01.928 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.010 seconds (success: True)
2025-07-12 17:56:03.929 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.005 seconds (success: True)
2025-07-12 17:56:05.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:56:07.918 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:56:09.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:56:11.919 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:56:13.918 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 17:56:15.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-12 17:56:17.921 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:56:19.920 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 17:56:21.918 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)

inventor7777 commented:
I guess “ 2025-07-12 17:55:04.053 ERROR (MainThread) [homeassistant.components.mjpeg] Error getting new camera image from Centauri Carbon Camera: All connection attempts failed” could be the error but it’s saying camera image not cover image. That could be an improvement you could make, stop it trying to pull the camera view when not in proxy mode. (Or maybe it isn’t and I just can’t interpret this log)

danielcherubini commented:
Camera Image, is not the same thing as cover image or thumbnail.. totally different things.

BUT... looking at your logs, you're not getting any errors when fetching the image, so its not an issue getting the image.. im going to add more debug logs

inventor7777 commented:
Yeah that’s why I was saying you could disable camera image polling if proxy isn’t enabled.

but what is thumbnail va camera vs cover image?

cover image is the gcode / 3d preview i know that and camera is the video feed but what is thumbnail then?

and you said earlier that I need to be looking for a thumbnail? So please confirm what you mean. @danielcherubini

danielcherubini commented:
I am only focused on one issue at a time.. so thats great about the camera image, but thats a later problmem

Thumbnail/Cover Image is a bitmap graphic of your current print, not whats coming from inside the chamber, but the image of the print from the slicer. its called thumbnail in the printer, and Cover Image in home assistant, i use both interchangebly.

Debug logs are the ones you've been pasting, where you enable debug logging, vs the other logs, where i only log errors/warnings which is the standard home assistant logs

Settings > System > Logs

Thats where any actual errors get logged.. you do not want anything other than warning/error there.

Also

https://github.com/danielcherubini/elegoo-homeassistant/releases/tag/1.4.6b8

try this, and then look at the debug logs.. its going to be pretty verbose.

inventor7777 commented:
I have been going to system —> logs but with debug enabled. Is that right?

I will install the update once it shows up in hacs @danielcherubini

inventor7777 commented:
It’s not showing up in the normal GitHub releases or HACS without that link you sent. Do I just have to wait?

inventor7777 commented:
2025-07-12 18:27:59.011 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration mbapi2020 which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2025-07-12 18:27:59.016 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration xsense which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2025-07-12 18:27:59.019 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration parcelapp which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2025-07-12 18:27:59.021 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration elegoo_printer which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2025-07-12 18:27:59.024 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration bhyve which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2025-07-12 18:27:59.027 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration goveelife which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2025-07-12 18:27:59.030 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2025-07-12 18:27:59.032 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration moonraker which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2025-07-12 18:28:17.249 DEBUG (MainThread) [custom_components.elegoo_printer] CONFIGURATION {'brand': 'ELEGOO', 'connection': '979d4C788A4a78bC777A870F1A02867A', 'firmware': 'V1.1.29', 'id': '0c6817140107202800007c0000000000', 'ip_address': '192.168.50.123', 'model': 'Centauri Carbon', 'name': 'Centauri Carbon', 'printer_type': 'fdm', 'protocol': 'V3.0.0', 'proxy_enabled': False}
2025-07-12 18:28:17.251 DEBUG (MainThread) [custom_components.elegoo_printer] Connecting to printer: Centauri Carbon at 192.168.50.123 with proxy enabled False and printer.proxy: False
2025-07-12 18:28:17.252 INFO (MainThread) [custom_components.elegoo_printer] Closing connection to printer
2025-07-12 18:28:17.252 DEBUG (MainThread) [custom_components.elegoo_printer] Connecting to printer: Centauri Carbon at 192.168.50.123 proxy_enabled: False
2025-07-12 18:28:17.252 INFO (MainThread) [custom_components.elegoo_printer] Client connecting to WebSocket at: ws://192.168.50.123:3030/websocket
2025-07-12 18:28:18.754 INFO (MainThread) [custom_components.elegoo_printer] Client successfully connected to: Centauri Carbon, via proxy: False
2025-07-12 18:28:18.755 INFO (MainThread) [custom_components.elegoo_printer] Polling Started
2025-07-12 18:28:21.605 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.850 seconds (success: True)
2025-07-12 18:28:21.901 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/__init__.py:2: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  __import__('pkg_resources').declare_namespace(__name__)

2025-07-12 18:28:27.567 WARNING (MainThread) [custom_components.mbapi2020.websocket] MB-API access blocked. (First Message, expect a re-login) 429, message='Invalid response status', url='wss://websocket.amap-prod.mobilesdk.mercedes-benz.com/v2/ws', retry in 10 seconds...
2025-07-12 18:28:27.567 WARNING (MainThread) [custom_components.mbapi2020.websocket] 429 detected: Trying relogin with stored password for user beckybaird7@icloud.com
2025-07-12 18:28:34.404 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.29.5 is exactly one major version older than the runtime version 6.31.1 at api.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:28:34.413 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.29.5 is exactly one major version older than the runtime version 6.31.1 at api_options.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:28:34.611 WARNING (MainThread) [homeassistant.components.matter] Detected a device that has been reported to have firmware issues with light transitions. Transitions will be disabled for this light
2025-07-12 18:28:39.855 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 23 sensor entities
2025-07-12 18:28:39.944 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 1 image entities
2025-07-12 18:28:39.961 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 1 light entities
2025-07-12 18:28:39.965 DEBUG (MainThread) [custom_components.elegoo_printer] Adding 3 button entities
2025-07-12 18:28:40.785 WARNING (MainThread) [custom_components.mbapi2020.websocket] WSS Connection blocked: 429, message='Invalid response status', url='wss://websocket.amap-prod.mobilesdk.mercedes-benz.com/v2/ws', retry in 10 seconds...
2025-07-12 18:28:42.012 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.155 seconds (success: True)
2025-07-12 18:28:43.999 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.004 seconds (success: True)
2025-07-12 18:28:45.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 18:28:47.758 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 18:28:49.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 18:28:50.868 WARNING (MainThread) [custom_components.mbapi2020.websocket] WSS Connection blocked: 429, message='Invalid response status', url='wss://websocket.amap-prod.mobilesdk.mercedes-benz.com/v2/ws', retry in 40 seconds...
2025-07-12 18:28:51.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 18:28:53.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 18:28:55.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 18:28:57.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 18:28:59.758 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 18:29:00.061 WARNING (MainThread) [custom_components.mbapi2020] Account is blocked. Reload will happen after unblock at midnight (GMT).
2025-07-12 18:29:02.139 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.382 seconds (success: True)
2025-07-12 18:29:04.058 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.092 seconds (success: True)
2025-07-12 18:29:05.972 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.153 seconds (success: True)
2025-07-12 18:29:06.396 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/ProtocolMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.419 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/AudioFadeMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.465 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/PlayerPath.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.486 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/Origin.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.500 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/DeviceInfoMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.522 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/Common.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.541 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/NowPlayingClient.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.565 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/NowPlayingPlayer.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.615 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/AudioFadeResponseMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.646 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/ClientUpdatesConfigMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.674 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/ConfigureConnectionMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.711 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/CryptoPairingMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.734 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/GenericMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.747 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/GetKeyboardSessionMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.773 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/GetRemoteTextInputSessionMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.803 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/GetVolumeMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.826 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/GetVolumeResultMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:06.854 WARNING (ImportExecutor_0) [py.warnings] /usr/local/lib/python3.13/site-packages/google/protobuf/runtime_version.py:98: UserWarning: Protobuf gencode version 5.28.1 is exactly one major version older than the runtime version 6.31.1 at pyatv/protocols/mrp/protobuf/KeyboardMessage.proto. Please update the gencode to avoid compatibility violations in the next runtime release.
  warnings.warn(

2025-07-12 18:29:07.846 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.022 seconds (success: True)
2025-07-12 18:29:09.774 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.013 seconds (success: True)
2025-07-12 18:29:12.747 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.986 seconds (success: True)
2025-07-12 18:29:14.827 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.134 seconds (success: True)
2025-07-12 18:29:18.792 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.021 seconds (success: True)
2025-07-12 18:29:22.766 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.009 seconds (success: True)
2025-07-12 18:29:24.531 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [281473330805344] Error handling message: Unknown error (unknown_error) Gabriel Baird from fe80::18a5:c5b2:56cd:773a (Mozilla/5.0 (iPad; CPU OS 19_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2025.6 (io.robbie.HomeAssistant; build:2025.1305; iPadOS 26.0.0) Mobile/HomeAssistant, like Safari)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 28, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 599, in handle_manifest_list
    raise int_or_exc
homeassistant.loader.IntegrationNotFound: Integration 'ble_monitor' not found.
2025-07-12 18:29:26.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 18:29:28.246 DEBUG (MainThread) [custom_components.elegoo_printer] Fetching new image from printer
2025-07-12 18:29:28.247 DEBUG (MainThread) [custom_components.elegoo_printer] Task: None
2025-07-12 18:29:28.261 DEBUG (MainThread) [custom_components.elegoo_printer] No new image found
2025-07-12 18:29:28.326 ERROR (MainThread) [homeassistant.components.mjpeg] Error getting new camera image from Centauri Carbon Camera: All connection attempts failed
2025-07-12 18:29:30.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.002 seconds (success: True)
2025-07-12 18:29:31.122 WARNING (MainThread) [custom_components.mbapi2020.websocket] WSS Connection blocked: 429, message='Invalid response status', url='wss://websocket.amap-prod.mobilesdk.mercedes-benz.com/v2/ws', retry in 90 seconds...
2025-07-12 18:29:34.759 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.001 seconds (success: True)
2025-07-12 18:29:38.764 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.006 seconds (success: True)
2025-07-12 18:29:42.760 DEBUG (MainThread) [custom_components.elegoo_printer] Finished fetching elegoo_printer data in 2.003 seconds (success: True)
2025-07-12 18:29:44.336 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [281473330805344] Error handling message: Unknown error (unknown_error) Gabriel Baird from fe80::18a5:c5b2:56cd:773a (Mozilla/5.0 (iPad; CPU OS 19_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Home Assistant/2025.6 (io.robbie.HomeAssistant; build:2025.1305; iPadOS 26.0.0) Mobile/HomeAssistant, like Safari)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 28, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 599, in handle_manifest_list
    raise int_or_exc
homeassistant.loader.IntegrationNotFound: Integration 'ble_monitor' not found.

danielcherubini commented:
```
2025-07-12 18:29:28.246 DEBUG (MainThread) [custom_components.elegoo_printer] Fetching new image from printer
2025-07-12 18:29:28.247 DEBUG (MainThread) [custom_components.elegoo_printer] Task: None
2025-07-12 18:29:28.261 DEBUG (MainThread) [custom_components.elegoo_printer] No new image found
```

There it is!

danielcherubini commented:
```python
    async def async_get_task(
        self, include_last_task: bool
    ) -> PrintHistoryDetail | None:
        if current_task := await self.client.async_get_printer_current_task():
            return current_task
        elif include_last_task:
            if last_task := await self.client.async_get_printer_last_task():
                return last_task

        return None
```

This is the function, now we're passing false to that include_last_task boolean, so if it cannot find the current task, it will return none.

```python
    async def async_get_printer_current_task(self) -> PrintHistoryDetail | None:
        """
        Asynchronously retrieves the current print task details from the printer.

        Returns:
            PrintHistoryDetail | None: The details of the current print task if available, otherwise None.
        """
        if task_id := self.printer_data.status.print_info.task_id:
            current_task = self.printer_data.print_history.get(task_id)
            if current_task is not None:
                return current_task
            else:
                await self.get_printer_task_detail([task_id])
                return self.printer_data.print_history.get(task_id)

        return None
```

This is get current task.. So what this is doing, is checking if you have a task_id in the current print that you're doing.. then it checks the printer history for that task, which is an in memory cache so we don't hammer the printer all the time. if thats in the cache, then we return the cached item, else, we go get it and return that.

Else.. if all that fails, we return none.

So you're getting None from this function.. I'm adding more debug logging

danielcherubini commented:
@inventor7777 THANKYOU SO MUCH HERE!

https://github.com/danielcherubini/elegoo-homeassistant/releases/tag/v1.4.6b9

inventor7777 commented:
@danielcherubini YES IT WORKS!!

not the highest quality ever but it’s working now! I’m guessing that’s a printer limitation anyway.

>THANKYOU SO MUCH HERE!
Always glad to help! I’m just as excited about getting this working as you are :)

inventor7777 commented:
![IMG_1718](https://github.com/user-attachments/assets/761cebdf-3a8d-4468-9c74-91702c26701f)

danielcherubini commented:
> @danielcherubini YES IT WORKS!!
> 
> not the highest quality ever but it’s working now! I’m guessing that’s a printer limitation anyway.
> 
> > THANKYOU SO MUCH HERE!
> 
> Always glad to help! I’m just as excited about getting this working as you are :)

FANTASTIC!

OK lets play a game... tell me if you can spot the bug?

```python
    async def async_get_printer_current_task(self) -> PrintHistoryDetail | None:
        """
        Asynchronously retrieves the current print task details from the printer.

        Returns:
            PrintHistoryDetail | None: The details of the current print task if available, otherwise None.
        """
        if task_id := self.printer_data.status.print_info.task_id:
            current_task = self.printer_data.print_history.get(task_id)
            if current_task is not None:
                return current_task
            else:
                await self.get_printer_task_detail([task_id])
                return self.printer_data.print_history.get(task_id)

        return None
```

Its here... i fixed it when pushing the last version.

don't scroll down












```python
                await self.get_printer_task_detail([task_id])
                return self.printer_data.print_history.get(task_id)
```

i should have returned that first line, not the second, i deleted the second line and returned the first.. see the function_get_printer_task_detail, returns the object we need, which also the history get also does.. it was stupid and insidious

danielcherubini commented:
Great i will wrap this up and release it.

inventor7777 commented:
Makes sense! 

I am not very good at all at interpreting unfamiliar code yet but I’m learning fast with all of the open source stuff I run. This was actually a good learning experience for me as well as I didn’t even know where to find logs before 😂 😂 😂 

but I’m glad it’s finally fixed and done! Moving along nicely.

danielcherubini commented:
@inventor7777 can you make a new issue for that camera one you've seen? thanks!

inventor7777 commented:
Ok will do
